tools/cellxgene_census_builder/tests/test_spatial.py

```diff
@@ -0,0 +1,235 @@
+            "--manifest",
+            str(spatial_manifest),
+            # "--no-validate",
+        ],
+        check=True,
+    )
+
+    return SpatialBuild(
+        census_dir / build_tag / "soma",
+        spatial_manifest,
+        blocklist,
+    )
+
+
+def test_spatial_build(spatial_build):
+    manifest = load_manifest(str(spatial_build.manifest_path), str(spatial_build.blocklist))
+    census = cellxgene_census.open_soma(uri=str(spatial_build.soma_path))
+    obs = census["census_spatial_sequencing"]["homo_sapiens"].obs.read().concat().to_pandas()
+    assert set(obs["dataset_id"].unique()) == {d.dataset_id for d in manifest}
+
+
+def _to_df(somadf: tiledbsoma.DataFrame) -> pd.DataFrame:
+    return somadf.read().concat().to_pandas()
+
+
+def test_locations(spatial_build):
+    census = cellxgene_census.open_soma(uri=str(spatial_build.soma_path))
+    exp = census["census_spatial_sequencing"]["homo_sapiens"]
+
+    manifest = load_manifest(str(spatial_build.manifest_path), str(spatial_build.blocklist))
+    obs = census["census_spatial_sequencing"]["homo_sapiens"].obs.read().concat().to_pandas()
+    assay_types = obs[["dataset_id", "assay"]].drop_duplicates().set_index("dataset_id")["assay"]
+    h5ads = {d.dataset_id: Path(d.dataset_asset_h5ad_uri) for d in manifest}
+
+    obs_spatial_presence = _to_df(exp["obs_spatial_presence"])
+
+    # Test that the obs_spatial_presence join id + dataset work with locations
+    for scene_id, subdf in obs_spatial_presence.groupby("scene_id"):
+        assay = assay_types.loc[scene_id]
+        sdata = exp.axis_query(
+            "RNA", obs_query=tiledbsoma.AxisQuery(value_filter=f"dataset_id == '{scene_id}'")
+        ).to_spatialdata(X_name="raw")
+        locations_key = list(sdata.shapes.keys())[0]
+        locations = sdata.shapes[locations_key]
+
+        assert locations["soma_joinid"].isin(subdf["soma_joinid"]).all()
+        assert len(locations["radius"].unique()) == 1
+
+        # Check against values from original file
+        if assay == "Visium Spatial Gene Expression":
+            with h5py.File(h5ads[scene_id]) as f:
+                locations_orig = read_elem(f["obsm/spatial"])
+                library_id = list(filter(lambda x: x != "is_single", f["uns/spatial"].keys()))[0]
+                radius_orig = f[f"uns/spatial/{library_id}/scalefactors/spot_diameter_fullres"][()] / 2
+            np.testing.assert_allclose(
+                locations_orig, locations.sort_values("soma_joinid").get_coordinates(), atol=1e-6
+            )
+            np.testing.assert_allclose(radius_orig, locations["radius"].unique()[0], atol=1e-6)
+
+
+def test_no_normalized_matrix(spatial_build):
+    census = cellxgene_census.open_soma(uri=str(spatial_build.soma_path))
+    spatial = census["census_spatial_sequencing"]["homo_sapiens"]
+
+    assert ["raw"] == list(spatial.ms["RNA"]["X"].keys())
+
+
+def test_images(spatial_build):
+    # TODO: check the metadata of the images
```

Comment:
TODO

Reply:
The metadata I meant to check here was the spatial transformations. The relevant thing for that is the radius for the points which is actually checked in `test_locations` already